_delphi_utils_python/delphi_utils/logger.py

```diff
@@ -1,8 +1,20 @@
 """Structured logger utility for creating JSON logs in Delphi pipelines."""
 import logging
+import sys
 import structlog
 
-def get_structured_logger(name = __name__, filename = None):
+
+def handle_exceptions(logger):
+    """Handle exceptions using the provided logger."""
+    def exception_handler(etype, value, traceback):
+        logger.exception("Top-level exception occurred",
+                         exc_info=(etype, value, traceback))
+    sys.excepthook = exception_handler
```

Comment:
TIL!

are we likely to need `threading.excepthook` for the safegraph stuff?

Reply:
Good idea! Added.